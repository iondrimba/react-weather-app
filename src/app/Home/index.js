import React, { Component, Fragment } from 'react';
import Swiper from 'swiper/dist/js/swiper.js';
import ForeCast from '../../components/Forecast';
import ForeCastTemperature from '../../components/Forecast/Temperature';
import Location from '../../components/Location';
import Temperature from '../../components/Temperature';
import Navigation from '../../components/Navigation';
import RainProbability from '../../components/RainProbability';
import GPSLocation from '../../components/GPSLocation';
import Info from '../../components/Info';
import DateCurrent from '../../components/DateCurrent';
import PropTypes from 'prop-types';

class Home extends Component {
  constructor() {
    super();

    this.state = {
      currentForecast: 'hourly',
      forecastIndex: ['hourly', 'daily'],
    };

    this.onInfoClick = this.onInfoClick.bind(this);
    this.onInfoClose = this.onInfoClose.bind(this);
  }

  onInfoClick() {
    this.props.onInfoClick();
  }

  onInfoClose() {
    this.props.onInfoClose();
  }

  componentDidMount() {
    this.forecasts = [...document.querySelectorAll('.forecasts__period')];

    this.swiper = new Swiper('.swiper-container', {
      direction: 'horizontal',
      loop: false
    });

    this.swiper.on('slideChangeTransitionEnd', () => {
      this.setState({ currentForecast: this.state.forecastIndex[this.swiper.activeIndex] });
    });
  }

  render() {
    return <Fragment>
      <GPSLocation onGetCurrentLocation={this.props.onGetCurrentLocation} />
      <Info onInfoClick={this.onInfoClick} onInfoClose={this.onInfoClose} />
      <Location location={this.props.currentCondition.location} />
      <DateCurrent date={this.props.currentCondition.date} />
      <Temperature weather={this.props.currentCondition.weather} temperature={this.props.currentCondition.temperature} />

      <section className="forecasts">
        <Navigation currentForecast={this.state.currentForecast} />

        <div className="forecasts__scroll-panel swiper-container">
          <div className="swiper-wrapper">
            <section className="forecasts__period swiper-slide">
              {
                this.props.foreCastHourly.map((item, index) => <ForeCast key={`hourly-${index}`} id={`hourly-${index}`} title={`${item.time}:00`} icon={item.icon} animationDelay={index}>
                  <ForeCastTemperature temperature={item.temperature} />
                  <RainProbability probability={item.rainProbability} />
                </ForeCast>)
              }
            </section>

            <section className="forecasts__period forecasts--daily swiper-slide">
              {
                this.props.foreCastDaily.map((item, index) => <ForeCast key={`daily-${index}`} id={`daily-${index}`} title={item.weekDay} icon={item.icon} animationDelay={index}>
                  <div>
                    <ForeCastTemperature temperature={item.temperature.max} />
                    <ForeCastTemperature temperature={item.temperature.min} />
                  </div>
                  <RainProbability probability={item.rainProbability} />
                </ForeCast>)
              }
            </section>
          </div>
        </div>
      </section>
    </Fragment>
  }
}

Home.propTypes = {
  foreCastHourly: PropTypes.array,
  foreCastDaily: PropTypes.array,
  currentCondition: PropTypes.object,
  onGetCurrentLocation: PropTypes.func,
  onInfoClick: PropTypes.func,
  onInfoClose: PropTypes.func
};

export default Home;
