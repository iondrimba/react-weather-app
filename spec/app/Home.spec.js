import React from 'react';
import Home from '../../src/app/Home';
import { enzymeConfig, mount } from '../enzymeConfig';

enzymeConfig();

const currentCondition = {
  "currentCondition": {
    "location": "Rio",
    "temperature": 23,
    "date": "--",
    "weather": "Clear",
    "dayTime": false,
    "humidity": 30,
    "dataLoaded": false
  },
  "foreCastDaily": [
    {
      "weekDay": "mon",
      "rainProbability": 47,
      "icon": "wi-day-sunny",
      "temperature": {
        "max": 23,
        "min": 12
      }
    },
    {
      "weekDay": "tue",
      "rainProbability": 35,
      "icon": "wi-day-sunny",
      "temperature": {
        "max": 33,
        "min": 14
      }
    },
    {
      "weekDay": "wed",
      "rainProbability": 1,
      "icon": "wi-day-sunny",
      "temperature": {
        "max": 22,
        "min": 21
      }
    },
    {
      "weekDay": "Thu",
      "rainProbability": 10,
      "icon": "wi-day-sunny",
      "temperature": {
        "max": 30,
        "min": 25
      }
    },
    {
      "weekDay": "Fry",
      "rainProbability": 20,
      "icon": "wi-fog",
      "temperature": {
        "max": 19,
        "min": 11
      }
    }
  ],
  "foreCastHourly": [
    {
      "time": 8,
      "rainProbability": 20,
      "temperature": 23,
      "icon": "wi-fog"
    },
    {
      "time": 9,
      "rainProbability": 22,
      "temperature": 25,
      "icon": "wi-day-sunny"
    },
    {
      "time": 10,
      "rainProbability": 40,
      "temperature": 15,
      "icon": "wi-day-sunny"
    },
    {
      "time": 11,
      "rainProbability": 15,
      "temperature": 16,
      "icon": "wi-day-sunny"
    },
    {
      "time": 12,
      "rainProbability": 5,
      "temperature": 24,
      "icon": "wi-day-sunny"
    }
  ],
  "location": "SÃ£o Paulo",
  "date": "Monday, June 25, 2018",
  "temperature": 20,
  "weather": "Clear"
};

const foreCastDaily = [
  {
    "weekDay": "tue",
    "rainProbability": 0,
    "icon": "wi-night-alt-cloudy",
    "temperature": {
      "max": 26,
      "min": 16
    }
  },
  {
    "weekDay": "wed",
    "rainProbability": 13,
    "icon": "wi-day-cloudy",
    "temperature": {
      "max": 26,
      "min": 15
    }
  },
  {
    "weekDay": "thu",
    "rainProbability": 0,
    "icon": "wi-day-cloudy",
    "temperature": {
      "max": 26,
      "min": 15
    }
  },
  {
    "weekDay": "fry",
    "rainProbability": 3,
    "icon": "wi-day-sunny",
    "temperature": {
      "max": 26,
      "min": 15
    }
  },
  {
    "weekDay": "sat",
    "rainProbability": 4,
    "icon": "wi-night-alt-cloudy",
    "temperature": {
      "max": 26,
      "min": 14
    }
  }
]

const foreCastHourly = [
  {
    "time": 22,
    "rainProbability": 0,
    "temperature": 21,
    "icon": "wi-night-clear"
  },
  {
    "time": 23,
    "rainProbability": 0,
    "temperature": 20,
    "icon": "wi-night-clear"
  },
  {
    "time": "00",
    "rainProbability": 0,
    "temperature": 19,
    "icon": "wi-night-clear"
  },
  {
    "time": "01",
    "rainProbability": 0,
    "temperature": 19,
    "icon": "wi-night-clear"
  },
  {
    "time": "02",
    "rainProbability": 0,
    "temperature": 18,
    "icon": "wi-night-clear"
  }
]

describe('Home', () => {
  it('matches snapshot', () => {
    const component = mount(<Home currentCondition={currentCondition} foreCastHourly={foreCastHourly} foreCastDaily={foreCastDaily} />);

    expect(component).toMatchSnapshot();
  });
});
