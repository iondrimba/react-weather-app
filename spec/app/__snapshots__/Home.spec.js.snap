// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<Home
  currentCondition={
    Object {
      "currentCondition": Object {
        "dataLoaded": false,
        "date": "--",
        "dayTime": false,
        "humidity": 30,
        "location": "Rio",
        "temperature": 23,
        "weather": "Clear",
      },
      "date": "Monday, June 25, 2018",
      "foreCastDaily": Array [
        Object {
          "icon": "wi-day-sunny",
          "rainProbability": 47,
          "temperature": Object {
            "max": 23,
            "min": 12,
          },
          "weekDay": "mon",
        },
        Object {
          "icon": "wi-day-sunny",
          "rainProbability": 35,
          "temperature": Object {
            "max": 33,
            "min": 14,
          },
          "weekDay": "tue",
        },
        Object {
          "icon": "wi-day-sunny",
          "rainProbability": 1,
          "temperature": Object {
            "max": 22,
            "min": 21,
          },
          "weekDay": "wed",
        },
        Object {
          "icon": "wi-day-sunny",
          "rainProbability": 10,
          "temperature": Object {
            "max": 30,
            "min": 25,
          },
          "weekDay": "Thu",
        },
        Object {
          "icon": "wi-fog",
          "rainProbability": 20,
          "temperature": Object {
            "max": 19,
            "min": 11,
          },
          "weekDay": "Fry",
        },
      ],
      "foreCastHourly": Array [
        Object {
          "icon": "wi-fog",
          "rainProbability": 20,
          "temperature": 23,
          "time": 8,
        },
        Object {
          "icon": "wi-day-sunny",
          "rainProbability": 22,
          "temperature": 25,
          "time": 9,
        },
        Object {
          "icon": "wi-day-sunny",
          "rainProbability": 40,
          "temperature": 15,
          "time": 10,
        },
        Object {
          "icon": "wi-day-sunny",
          "rainProbability": 15,
          "temperature": 16,
          "time": 11,
        },
        Object {
          "icon": "wi-day-sunny",
          "rainProbability": 5,
          "temperature": 24,
          "time": 12,
        },
      ],
      "location": "São Paulo",
      "temperature": 20,
      "weather": "Clear",
    }
  }
  foreCastDaily={
    Array [
      Object {
        "icon": "wi-night-alt-cloudy",
        "rainProbability": 0,
        "temperature": Object {
          "max": 26,
          "min": 16,
        },
        "weekDay": "tue",
      },
      Object {
        "icon": "wi-day-cloudy",
        "rainProbability": 13,
        "temperature": Object {
          "max": 26,
          "min": 15,
        },
        "weekDay": "wed",
      },
      Object {
        "icon": "wi-day-cloudy",
        "rainProbability": 0,
        "temperature": Object {
          "max": 26,
          "min": 15,
        },
        "weekDay": "thu",
      },
      Object {
        "icon": "wi-day-sunny",
        "rainProbability": 3,
        "temperature": Object {
          "max": 26,
          "min": 15,
        },
        "weekDay": "fry",
      },
      Object {
        "icon": "wi-night-alt-cloudy",
        "rainProbability": 4,
        "temperature": Object {
          "max": 26,
          "min": 14,
        },
        "weekDay": "sat",
      },
    ]
  }
  foreCastHourly={
    Array [
      Object {
        "icon": "wi-night-clear",
        "rainProbability": 0,
        "temperature": 21,
        "time": 22,
      },
      Object {
        "icon": "wi-night-clear",
        "rainProbability": 0,
        "temperature": 20,
        "time": 23,
      },
      Object {
        "icon": "wi-night-clear",
        "rainProbability": 0,
        "temperature": 19,
        "time": "00",
      },
      Object {
        "icon": "wi-night-clear",
        "rainProbability": 0,
        "temperature": 19,
        "time": "01",
      },
      Object {
        "icon": "wi-night-clear",
        "rainProbability": 0,
        "temperature": 18,
        "time": "02",
      },
    ]
  }
  lastUpdate="00:00"
  updating={false}
>
  <GPSLocation>
    <button
      aria-label="Get current location"
      className="gps-location"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <svg
        id="Layer_1"
        viewBox="0 0 96 96"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="XMLID_1_"
        >
          <circle
            cx="48"
            cy="48"
            id="XMLID_3_"
            r="9.8"
          />
          <path
            d="M80.5 44.8h-6.7C72.3 33 63 23.7 51.3 22.2v-6.7h-6.5v6.7C33 23.7 23.7 33 22.2 44.8h-6.7v6.5h6.7C23.7 63 33 72.3 44.8 73.8v6.7h6.5v-6.7C63 72.3 72.3 63 73.8 51.3h6.7v-6.5zM48 67.5c-10.8 0-19.5-8.7-19.5-19.5S37.2 28.5 48 28.5 67.5 37.2 67.5 48 58.8 67.5 48 67.5z"
            id="XMLID_6_"
          />
        </g>
      </svg>
    </button>
  </GPSLocation>
  <Info>
    <button
      aria-label="Information"
      className="button-info"
      type="button"
    >
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 0C5.371 0 0 5.371 0 12s5.371 12 12 12 12-5.371 12-12S18.629 0 12 0zm0 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 3.813c-.184 0-.336-.004-.5.03a1.127 1.127 0 0 0-.438.22.98.98 0 0 0-.28.375c-.071.152-.095.332-.095.562 0 .227.024.406.094.563.07.156.16.28.281.375.122.093.274.148.438.187.164.04.316.063.5.063.18 0 .371-.024.531-.063.16-.04.285-.094.406-.188a.951.951 0 0 0 .282-.375c.07-.152.125-.335.125-.562 0-.23-.055-.41-.125-.563a.98.98 0 0 0-.281-.375c-.122-.093-.247-.183-.407-.218-.16-.035-.351-.032-.531-.032zm-1.219 3.343v8.969h2.438V9.156z"
        />
      </svg>
    </button>
  </Info>
  <Location
    location="São Paulo"
  >
    <section
      className="location"
    >
      <span
        className="location__icon"
      >
        <svg
          enableBackground="new 0 0 24 24"
          height="24px"
          id="Layer_1"
          version="1.1"
          viewBox="0 0 24 24"
          width="24px"
        >
          <path
            d="M11.513,12.38c-2.117,0-3.835-1.729-3.835-3.862c0-2.135,1.718-3.863,3.835-3.863s3.835,1.729,3.835,3.863  C15.348,10.65,13.63,12.38,11.513,12.38 M11.513,0C6.825,0,3.025,3.827,3.025,8.549c0,4.46,3.844,10.213,6.411,13.014  c0.959,1.045,2.076,2.454,2.076,2.454s1.2-1.417,2.229-2.493C16.306,18.84,20,13.451,20,8.549C20,3.827,16.2,0,11.513,0"
          />
        </svg>
      </span>
      <span
        className="location__text"
      >
        São Paulo
      </span>
    </section>
  </Location>
  <DateCurrent
    date="Monday, June 25, 2018"
  >
    <section
      className="date-current"
    >
      <span
        className="date-current__text"
      >
        Monday, June 25, 2018
      </span>
    </section>
  </DateCurrent>
  <Temperature
    temperature={20}
    weather="Clear"
  >
    <section
      className="current-condition"
    >
      <div
        className="wrapper-temperature"
      >
        <span
          className="temperature__status"
        >
          Clear
        </span>
        <div
          className="temperature"
        >
          <span
            className="temperature__value"
          >
            20
          </span>
          <div
            className="temperature__unit"
          >
            <span
              className="temperature__unit-dot"
            />
            <span
              className="temperature__unit-letter"
            >
              c
            </span>
          </div>
        </div>
      </div>
    </section>
  </Temperature>
  <Refresh
    time="00:00"
    updating={false}
  >
    <section
      className="refresh  "
    >
      <span
        className="refresh__time"
      >
        00:00
      </span>
      <button
        aria-label="Refresh content"
        className="button-refresh"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        <svg
          viewBox="0 0 96 96"
        >
          <path
            d="M48 15.5c-8.8 0-17 3.5-23 9.5l-6.3-6.3v19.5h19.5l-8.6-8.6C34.4 24.8 40.9 22 48 22c14.3 0 26 11.7 26 26S62.3 74 48 74 22 62.3 22 48h-6.5c0 17.9 14.6 32.5 32.5 32.5S80.5 65.9 80.5 48 65.9 15.5 48 15.5z"
          />
        </svg>
      </button>
    </section>
  </Refresh>
  <section
    className="forecasts"
  >
    <div
      className="forecasts__scroll-panel swiper-container"
    >
      <div
        className="swiper-wrapper"
      >
        <section
          className="forecasts__period swiper-slide"
        >
          <Forecast
            animationDelay={0}
            icon="wi-night-clear"
            id="hourly-0"
            key="hourly-0"
            title="22:00"
          >
            <div
              className="forecast"
              data-id="hourly-0"
            >
              <span
                className="forecast__title"
              >
                22:00
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-night-clear.svg"
              />
              <ForeCastTemperature
                temperature={21}
              >
                <p
                  className="forecast__temperature"
                >
                  21
                  <span
                    className="temperature__unit-dot"
                  />
                </p>
              </ForeCastTemperature>
              <RainProbability
                probability={0}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    0%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
          <Forecast
            animationDelay={1}
            icon="wi-night-clear"
            id="hourly-1"
            key="hourly-1"
            title="23:00"
          >
            <div
              className="forecast"
              data-id="hourly-1"
            >
              <span
                className="forecast__title"
              >
                23:00
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-night-clear.svg"
              />
              <ForeCastTemperature
                temperature={20}
              >
                <p
                  className="forecast__temperature"
                >
                  20
                  <span
                    className="temperature__unit-dot"
                  />
                </p>
              </ForeCastTemperature>
              <RainProbability
                probability={0}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    0%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
          <Forecast
            animationDelay={2}
            icon="wi-night-clear"
            id="hourly-2"
            key="hourly-2"
            title="00:00"
          >
            <div
              className="forecast"
              data-id="hourly-2"
            >
              <span
                className="forecast__title"
              >
                00:00
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-night-clear.svg"
              />
              <ForeCastTemperature
                temperature={19}
              >
                <p
                  className="forecast__temperature"
                >
                  19
                  <span
                    className="temperature__unit-dot"
                  />
                </p>
              </ForeCastTemperature>
              <RainProbability
                probability={0}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    0%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
          <Forecast
            animationDelay={3}
            icon="wi-night-clear"
            id="hourly-3"
            key="hourly-3"
            title="01:00"
          >
            <div
              className="forecast"
              data-id="hourly-3"
            >
              <span
                className="forecast__title"
              >
                01:00
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-night-clear.svg"
              />
              <ForeCastTemperature
                temperature={19}
              >
                <p
                  className="forecast__temperature"
                >
                  19
                  <span
                    className="temperature__unit-dot"
                  />
                </p>
              </ForeCastTemperature>
              <RainProbability
                probability={0}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    0%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
          <Forecast
            animationDelay={4}
            icon="wi-night-clear"
            id="hourly-4"
            key="hourly-4"
            title="02:00"
          >
            <div
              className="forecast"
              data-id="hourly-4"
            >
              <span
                className="forecast__title"
              >
                02:00
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-night-clear.svg"
              />
              <ForeCastTemperature
                temperature={18}
              >
                <p
                  className="forecast__temperature"
                >
                  18
                  <span
                    className="temperature__unit-dot"
                  />
                </p>
              </ForeCastTemperature>
              <RainProbability
                probability={0}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    0%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
        </section>
        <section
          className="forecasts__period forecasts--daily swiper-slide"
        >
          <Forecast
            animationDelay={0}
            icon="wi-night-alt-cloudy"
            id="daily-0"
            key="daily-0"
            title="tue"
          >
            <div
              className="forecast"
              data-id="daily-0"
            >
              <span
                className="forecast__title"
              >
                tue
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-night-alt-cloudy.svg"
              />
              <div>
                <ForeCastTemperature
                  temperature={26}
                >
                  <p
                    className="forecast__temperature"
                  >
                    26
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
                <ForeCastTemperature
                  temperature={16}
                >
                  <p
                    className="forecast__temperature"
                  >
                    16
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
              </div>
              <RainProbability
                probability={0}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    0%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
          <Forecast
            animationDelay={1}
            icon="wi-day-cloudy"
            id="daily-1"
            key="daily-1"
            title="wed"
          >
            <div
              className="forecast"
              data-id="daily-1"
            >
              <span
                className="forecast__title"
              >
                wed
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-day-cloudy.svg"
              />
              <div>
                <ForeCastTemperature
                  temperature={26}
                >
                  <p
                    className="forecast__temperature"
                  >
                    26
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
                <ForeCastTemperature
                  temperature={15}
                >
                  <p
                    className="forecast__temperature"
                  >
                    15
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
              </div>
              <RainProbability
                probability={13}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    13%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
          <Forecast
            animationDelay={2}
            icon="wi-day-cloudy"
            id="daily-2"
            key="daily-2"
            title="thu"
          >
            <div
              className="forecast"
              data-id="daily-2"
            >
              <span
                className="forecast__title"
              >
                thu
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-day-cloudy.svg"
              />
              <div>
                <ForeCastTemperature
                  temperature={26}
                >
                  <p
                    className="forecast__temperature"
                  >
                    26
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
                <ForeCastTemperature
                  temperature={15}
                >
                  <p
                    className="forecast__temperature"
                  >
                    15
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
              </div>
              <RainProbability
                probability={0}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    0%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
          <Forecast
            animationDelay={3}
            icon="wi-day-sunny"
            id="daily-3"
            key="daily-3"
            title="fry"
          >
            <div
              className="forecast"
              data-id="daily-3"
            >
              <span
                className="forecast__title"
              >
                fry
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-day-sunny.svg"
              />
              <div>
                <ForeCastTemperature
                  temperature={26}
                >
                  <p
                    className="forecast__temperature"
                  >
                    26
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
                <ForeCastTemperature
                  temperature={15}
                >
                  <p
                    className="forecast__temperature"
                  >
                    15
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
              </div>
              <RainProbability
                probability={3}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    3%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
          <Forecast
            animationDelay={4}
            icon="wi-night-alt-cloudy"
            id="daily-4"
            key="daily-4"
            title="sat"
          >
            <div
              className="forecast"
              data-id="daily-4"
            >
              <span
                className="forecast__title"
              >
                sat
              </span>
              <img
                alt="icon"
                className="forecast__icon"
                src="svg/wi-night-alt-cloudy.svg"
              />
              <div>
                <ForeCastTemperature
                  temperature={26}
                >
                  <p
                    className="forecast__temperature"
                  >
                    26
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
                <ForeCastTemperature
                  temperature={14}
                >
                  <p
                    className="forecast__temperature"
                  >
                    14
                    <span
                      className="temperature__unit-dot"
                    />
                  </p>
                </ForeCastTemperature>
              </div>
              <RainProbability
                probability={4}
              >
                <div
                  className="rain"
                >
                  <img
                    alt="alt"
                    className="rain__icon"
                    height="24"
                    src="/svg/rain-probality.svg"
                    width="24"
                  />
                  <span
                    className="rain__probability"
                  >
                    4%
                  </span>
                </div>
              </RainProbability>
            </div>
          </Forecast>
        </section>
      </div>
    </div>
    <Navigation
      currentForecast="hourly"
    >
      <section
        className="navigation hourly"
      >
        <div
          className="navigation__dot"
        />
        <div
          className="navigation__dot"
        />
      </section>
    </Navigation>
  </section>
</Home>
`;
